python/dgl/graphbolt/impl/neighbor_sampler.py

```diff
@@ -277,11 +337,22 @@ def _set_input_nodes(minibatch):
 
     # pylint: disable=arguments-differ
     def sampling_stages(
-        self, datapipe, graph, fanouts, replace, prob_name, deduplicate, sampler
+        self,
+        datapipe,
+        graph,
+        fanouts,
+        replace,
+        prob_name,
+        deduplicate,
+        sampler,
+        layer_dependency,
     ):
         datapipe = datapipe.transform(
             partial(self._prepare, graph.node_type_to_id)
         )
+        is_labor = sampler.__name__ == "sample_layer_neighbors"
+        if is_labor:
+            datapipe = datapipe.transform(self._set_seed)
```

Comment:
Same here, is it possible to move this sample_layer_neighbors specific code to its own class, instead of hack in the parent class.

Reply:
How do you propose we do that without replicating the whole implementation?

Reply:
I understand that. But I feel anti-oop pattern is better than replicating the code. I plan to base Temporal samplers into the NeighborSamplerImpl as well, so that the GPU sampling optimizations can be enabled for the temporal case as well.